esmvalcore/preprocessor/_time.py

```diff
@@ -456,6 +461,26 @@ def anomalies(cube, period):
         data[indexes] = data[indexes] - ref[cube_coord.points[i]]
 
     cube = cube.copy(data)
+
+    # Standardize the results if requested
+    if standardize:
+        cube_stddev = climate_statistics(cube,
+                                         operator='std_dev',
+                                         period=period)
+        assert cube.ndim == cube_stddev.ndim
+        ratio = [i / j for i, j in
+                 zip(cube.shape, cube_stddev.shape)]
+        # This will raise an error if the length of the cube their time axes
+        # are not multiples of each other or if other shapes are not equal
+        if not all([ratio[0] % 1 == 0] + [i == 1 for i in ratio[1:]]):
+            raise ValueError(
+                "Cannot safely apply preprocessor to this dataset,\
+                since the full time period of this dataset is not\
+                a multiple of the period {0}".format(period)
+            )
+        reps = tuple([int(i) for i in ratio])
+        cube.data = cube.data / np.tile(cube_stddev.data, reps)
```

Comment:
```suggestion
        cube.data = cube.core_data() / da.tile(cube_stddev.core_data(), reps)
```
Please try to keep the data lazy, maybe using something like the above?

Reply:
Tile is unfortunately not available in `dask`. [Here](https://stackoverflow.com/questions/38054970/what-is-the-dask-equivalent-of-numpy-tile) it is suggested to use concatenate, but it would be slightly more complicated in our case since `reps` is a tuple. Can we assume that the time axis always corresponds to index 0 ? In that case I can think of an implementation.

Reply:
Can't you simply use [`da.broadcast_to`](https://docs.dask.org/en/latest/array-api.html#dask.array.broadcast_to)?

Reply:
I.e. `cube.data = cube.core_data() / da.broadcast_to(cube_stddev.core_data(), cube.shape)`

Reply:
Actually broadcasting often fails here, so that is not a solution. But given the two checks above (which itself can be simplified a bit more, will think about this)  I am actually quite confident that concatenating along the first axis with given repetitions is safe. :)